presto-orc/src/main/java/com/facebook/presto/orc/writer/MapFlatColumnWriter.java

```diff
@@ -445,18 +451,21 @@ public void reset()
         rowsInFinishedRowGroups.clear();
         nonNullRowGroupValueCount = 0;
         nonNullStripeValueCount = 0;
+        rawSize = 0;
+        stripeRawSize = 0;
         Arrays.fill(valueWritersLastRow, 0);
     }
 
-    private MapFlatValueWriter createNewValueWriter(DwrfProto.KeyInfo dwrfKey)
+    private MapFlatValueWriter createNewValueWriter(DwrfProto.KeyInfo dwrfKey, int keyRawSize)
```

Comment:
Flat map writer needs to account for key raw size the same way the regular MapWriter does. Previously the flat map writer ignored the key raw size.

In case for integer based keys we use the fixed type width, for varchar/varbinary keys we use the size of that key.